### DISCUSSION TITLE
2011/04/18/labview%E4%B8%AD%E5%AE%9E%E7%8E%B0%E9%93%BE%E8%A1%A8%E3%80%81%E6%A0%91%E7%AD%89%E6%95%B0%E6%8D%AE%E7%BB%93%E6%9E%84

### DISCUSSION THREAD

**Discussion by giscus** [ruanqizhen/blog#326] [Announcements]
# 2011/04/18/labview%E4%B8%AD%E5%AE%9E%E7%8E%B0%E9%93%BE%E8%A1%A8%E3%80%81%E6%A0%91%E7%AD%89%E6%95%B0%E6%8D%AE%E7%BB%93%E6%9E%84

https://labview.qizhen.xyz/

https://blog.qizhen.xyz/2011/04/18/labview%E4%B8%AD%E5%AE%9E%E7%8E%B0%E9%93%BE%E8%A1%A8%E3%80%81%E6%A0%91%E7%AD%89%E6%95%B0%E6%8D%AE%E7%BB%93%E6%9E%84

**Comment by ruanqizhen** [comment]
之前博客下的留言：

lelewestlife
2011年06月27日 10:42 编辑
您好，我是您的《我和labview》这本书的忠实读者。最近有一个问题困扰着我，我想请教您一下。我是一名在读研究生，最近实验室有一个上位机平台的项目，用labview来做的，就是现实下位机传上来的实时数据，并且动态的存入数据库。现在面临的问题是，软件运行没有问题，但是运行时间长了之后，比如12个小时，内存使用就会增加300M到400M的样子。我看了程序，里面没有出现打开文件句柄而没有关闭的情况。我想问一下，这是其他什么原因导致的呢？局部变量的使用会导致内存泄露吗？非常感谢您的回答！

回复
ruanqizhen
2011年06月28日 17:14 编辑
局部变量应该不会引起内存泄漏。你可以自己调试一下，看泄漏出现在哪些代码中。
我一般用这种方法调试，先把一部分代码注释起来，看看内存泄露还在不在。然后再进一步注释掉少一些代码，一步一步找出泄漏。
还有就是，但看资源管理器增加了三四百兆，并不能确保是出现了泄漏。LabVIEW如果觉得你不需要更多的内存，可能会先占用一部分内存不放，等你的内存吃紧了再释放掉。

回复
Pingback： 《我和LabVIEW》补充内容和注释 « 凡人琐忆 编辑
nextpad
2011年04月19日 17:54 编辑
很赞！

回复

**Comment by ruanqizhen** [comment]
之前博客下的留言：

长生天
2011年05月1日 19:49 编辑
你的相片说明你年纪不小了。怎么孩子还这么小那！

回复
ruanqizhen
2011年05月3日 08:59 编辑
俺们大器晚成呗 :)

回复
Jiucheng Nie
2011年04月30日 04:10 编辑
您好，我是一名硕士在读生，请问，如果我想到NI公司从事LabView工程师的工作，应该具备哪些方面的知识呢？例如我需要掌握哪些编程语言或哪些软件的操作？请阮工指点一下，我好朝着这个方向努力

回复
ruanqizhen
2011年05月3日 08:55 编辑
可能你用到的每个面试官或者应聘的不同职位都有不同的要求。我觉得要给面试官留下好印象主要还不在于学过哪一门编程语言，而在于有没有把曾经学过的东西学精通。如果能把现在正在研究的课题搞得很透彻，面试就不怕了。

回复
Jiucheng Nie
2011年05月6日 23:51 编辑
谢谢您，当初本科毕业设计做的课题就是基于LabView的一套血压采集处理系统，然后读了您写的书，于是就对LabView的工作特别向往，非常感谢您的回复，我打算再把LabView学的更透彻一些，另外再把C和C++捡起来

回复
Jiucheng Nie
2011年04月30日 04:04 编辑
您好

回复
rabbitlee1982
2011年04月29日 22:02 编辑
还是幼儿园的最好看~

回复
淘宝女装推荐
2011年04月26日 16:48 编辑
哈哈，表情很淡定阿

回复
勺子
2011年04月24日 21:25 编辑
你LP說得很對挖。。。

回复
haiyingjia
2011年04月24日 19:35 编辑
你高中时长的比这张好多呢,顶多是盗窃犯吧

回复
